src/components/smart/Table.vue

```diff
@@ -1,69 +1,321 @@
 <template>
-  <div class="overflow-auto border shadow-md rounded-md border-dividerDark">
-    <table class="w-full">
-      <thead>
-        <slot name="head">
+  <div class="flex flex-1 flex-col">
+    <div v-if="pagination" class="mb-3 flex justify-end">
+      <div class="flex w-min">
+        <HoppButtonSecondary
+          outline
+          filled
+          :icon="IconLeft"
+          :disabled="page === 1"
+          @click="changePage(PageDirection.Previous)"
+        />
+
+        <div class="flex h-full w-10 items-center justify-center">
+          <p>{{ page }}</p>
+        </div>
+
+        <HoppButtonSecondary
+          outline
+          filled
+          :icon="IconRight"
+          :disabled="page === pagination.totalPages"
+          @click="changePage(PageDirection.Next)"
+        />
+      </div>
+    </div>
+
+    <div class="overflow-auto rounded-md border border-dividerDark shadow-md">
+      <div v-if="searchBar" class="flex w-full items-center bg-primary">
+        <icon-lucide-search class="mx-3 text-xs" />
+        <input
+          v-model="searchQuery"
+          class="h-full w-full bg-primary py-3"
+          :placeholder="searchBar.placeholder ?? 'Search...'"
+        />
+      </div>
+      <div v-if="isSpinnerEnabled" class="mx-auto my-3 h-5 w-5 text-center">
+        <HoppSmartSpinner />
+      </div>
+
+      <table v-else-if="list" class="w-full">
+        <thead v-if="list.length > 0">
```

Comment:
Shouldn't the table rendering depend on the length of the list and the check be applied on the `table` element straightaway?